### PR TITLE
Fix `No Cargo.toml!` error by normalizing DependencyFile names

### PR DESCRIPTION
**Pull request by thavaahariharangit**
### What are you trying to accomplish?

Fix the "No Cargo.toml!" error in Dependabot by normalizing DependencyFile names to remove leading slashes that cause filename mismatch between file fetcher and parser.

Fixing community issue: https://github.com/dependabot/dependabot-core/issues/13345

### Anything you want to highlight for special attention from reviewers?

- **Scoped fix:** Modified only `Cargo::FileFetcher` by overriding `load_cloned_file_if_present` and `fetch_file_from_host` to normalize filenames using `Pathname.new(file.name).cleanpath.to_s.gsub(%r{^/+}, "")`
- **Reviewer feedback addressed:** Changed from base class modification to Cargo-specific implementation per @robaiken'\''s comment
- **Edge case handling:** The fix preserves subdirectory paths while stripping only leading slashes
- **Code quality:** Compacted file to stay under RuboCop 350-line ClassLength limit without requiring exceptions
- **Test coverage:** Added comprehensive tests covering normalization logic, file fetching, and parser integration (14 new test cases)

### How will you know you've accomplished your goal?

 - The "No Cargo.toml!" error no longer occurs when processing the test repository `divviup/divviup-api`
- New unit tests pass, covering the specific bug case (`/Cargo.toml → Cargo.toml`) and edge cases
- File parser compatibility test confirms files with normalized names can be found by `get_original_file("Cargo.toml")`
- All existing Cargo tests (165 examples) continue to pass

Before fix:
```
updater | 2025/10/20 20:43:13 INFO Results:
Dependabot encountered '1' error(s) during execution, please check the logs for more details.
+-----------------------------------------------+
|                    Errors                     |
+---------------+-------------------------------+
| Type          | Details                       |
+---------------+-------------------------------+
| unknown_error | {                             |
|               |   "message": "No Cargo.toml!" |
|               | }                             |
+---------------+-------------------------------+
Failure running container 82b532cc7867e484617b548ff7a4d76df53da4b14d6e8812447b6d8f0c896354: Error: Command failed with exit code 1: /bin/sh -c $DEPENDABOT_HOME/dependabot-updater/bin/run update_files
```
Ref: https://github.com/divviup/divviup-api/actions/runs/18664383545/job/53212166360

After fix:
```
updater | 2025/10/22 13:15:11 INFO Finished job processing
updater | 2025/10/22 13:15:11 INFO Results:
updater | +----------------------------------------------------+
updater | |        Changes to Dependabot Pull Requests         |
updater | +---------+------------------------------------------+
updater | | created | janus_messages ( from 0.7.93 to 0.7.94 ) |
updater | | created | prometheus ( from 0.13.4 to 0.14.0 )     |
updater | | created | rand ( from 0.8.5 to 0.9.0 )             |
updater | | created | clap ( from 4.5.48 to 4.5.50 )           |
updater | +---------+------------------------------------------+
```

### Checklist

- [x] I have run the complete test suite to ensure all tests and linters pass.
- [x] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x] I have written clear and descriptive commit messages.
- [x] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x] I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

File: common/lib/dependabot/file_fetchers/base.rb [hunk 1]
```diff
@@ -241,7 +241,7 @@ def load_cloned_file_if_present(filename)
                end
 
         DependencyFile.new(
-          name: Pathname.new(filename).cleanpath.to_path,
+          name: Pathname.new(filename).cleanpath.to_s.gsub(%r{^/+}, ""),
```
Comment (robaiken) [inline common/lib/dependabot/file_fetchers/base.rb]: @thavaahariharangit since this issue only affects cargo, updating the base class of file_fetchers might be more than we need

**Comment by edmorley** [comment]
Hi! Please could this be expedited?

**Comment by reneleonhardt** [comment]
Is it normal that `ci (python, python, pip)` runs longer than 55 minutes so it will be killed?
https://github.com/dependabot/dependabot-core/actions/runs/19271753727/job/55101598173?pr=13359

> The hosted runner lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.
